docs/photutils/psf.rst

```diff
@@ -28,10 +28,379 @@ model-fitting analysis, regardless to exactly what kind of model is actually
 being fit.  We take this road, using "PSF photometry" as shorthand for the
 general approach.
 
+PSF Photometry in Crowded Fields
+--------------------------------
+
+Photutils provides an implementation of the DAOPHOT algorithm
```

Comment:
I would slightly rephrase this to say something like "Photutils provides a modular set of tools to perform PSF photometry for different science cases.  These are implemented as separate classes to do sub-tasks of PSF photometry.  It also provides high-level classes that connect these pieces together.  In particular, it contains an implementation of the DAOPHOT..." and then continue as is.

That is, get across the idea that it's not *just* a re-implementation of DAOPHOT, but a pluggable thing that be default is basically the DAOPHOT loop, but can be much more if you play around with it.